src/azure-cli/azure/cli/command_modules/acr/agentpool.py

```diff
@@ -0,0 +1,127 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from msrest.exceptions import ValidationError
+from knack.log import get_logger
+from knack.util import CLIError
+from ._utils import (
+    get_registry_by_name,
+    validate_managed_registry
+)
+
+
+DEFAULT_COUNT = 1
+DEFAULT_TIER = 'S1'
+DEFAULT_OS = 'Linux'
+
+logger = get_logger(__name__)
+
+def acr_agentpool_create(cmd,
+                         client,
+                         agent_pool_name,
+                         registry_name,
+                         resource_group_name=None,
+                         count=DEFAULT_COUNT,
+                         tier=DEFAULT_TIER,
+                         os_type=DEFAULT_OS,
+                         vnet_id=None):
+
+    registry, resource_group_name = get_registry_by_name(
+        cmd.cli_ctx, registry_name, resource_group_name)
+
+    AgentPool = cmd.get_models('AgentPool')
+
+    agentpool_create_paramters = AgentPool(
+        location=registry.location,
+        count=count,
+        tier=tier,
+        os=os_type,
+        VirtualNetworkSubnetResourceId=vnet_id
+    )
+
+    try:
+        return client.create(resource_group_name=resource_group_name,
+                             registry_name=registry_name,
+                             agent_pool_name=agent_pool_name,
+                             agent_pool=agentpool_create_paramters)
+    except ValidationError as e:
+        raise CLIError(e)
+
+
+def acr_agentpool_update(cmd,
+                         client,
+                         agent_pool_name,
+                         registry_name,
+                         resource_group_name=None,
+                         tags=None,
+                         count=None):
+
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+
+    AgentPoolUpdateParameters = cmd.get_models('AgentPoolUpdateParameters')
+
+    agentpool_create_paramters = AgentPoolUpdateParameters(
```

Comment:
nit: agentpool_update_parameters*

Reply:
What do you do with this object?

Reply:
remove this

Reply:
@huanwu  ^^

Reply:
I see. But we need remove the tag parameter from the update method.

Reply:
yes, the model is not being used I just wanted to have the discussion. Because it was weird to me that we have an update parameters but we have to pass the values raw to the Update function.
I left a comment on the PR exactly for this line. Shouldn't we pass the updateParameters model to the update function like in tasks? So if the future the capability of update increases we only modify the data model and not the update function parameters? To be more consistant and it looks cleaner to me as well.

Reply:
for example. Right now we want to remove tag from update. by doing so this might break the code because we're passing a tag value that is no longer needed but if we added a tag value to the datamodel it would just be ignored, and it will not break the code.